subprojects/packagefiles/mpark-patterns/meson.build

```diff
@@ -0,0 +1,26 @@
+project(
+    'MPark.Patterns',
+    'cpp',
+    default_options: ['cpp_std=c++17'],
+    license: 'BSL-1.0',
+    version: '0.3.0',
+    meson_version: '>=1.6.1',
+)
+
+if meson.is_subproject() and meson.get_compiler('cpp').get_id() == 'msvc'
+    add_global_arguments('/Zc:__cplusplus', language: 'cpp')
```

Comment:
You can't set global arguments when built as a subproject. You should use `add_project_arguments` instead.

Reply:
Meson will automatically inject `/Zc:__cplusplus` if the version of MSVC supports it, this shouldn't be necessary anyway.